src/transaction-runner.coffee

```diff
@@ -449,6 +417,49 @@ class TransactionRunner
         configuration.emitter.emit 'test error', error, test, () ->
         return callback()
 
+  validateTransaction: (test, transaction, callback) ->
+    configuration = @configuration
+
+    gavel.isValid transaction.real, transaction.expected, 'response', (isValidError, isValid) ->
+      if isValidError
+        configuration.emitter.emit 'test error', isValidError, test, () ->
+
+      test.start = test.start
+      test.title = transaction.id
+      test.actual = transaction.real
+      test.expected = transaction.expected
+      test.request = transaction.request
+
+      if isValid
+        test.status = "pass"
+      else
+        test.status = "fail"
+
+      gavel.validate transaction.real, transaction.expected, 'response', (validateError, result) ->
+        if not isValidError and validateError
+          configuration.emitter.emit 'test error', validateError, test, () ->
+
+        message = ''
+
+        for resultKey, data of result
+          if resultKey isnt 'version'
+            for entityResult in data['results']
+              message += resultKey + ": " + entityResult['message'] + "\n"
+
+        test.message = message
+        test.results = result
+        test.results['general'] = []
```

Comment:
should it be `test.results['general'] ?= []` ? Because before hooks might've added something there already?